### PR TITLE
EPMRPP-38542 || Nested step client side

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/annotations/Step.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.epam.reportportal.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/Step.java]: Please, give documentation for this annotation

File: src/main/java/com/epam/reportportal/annotations/StepTemplateConfig.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.epam.reportportal.annotations;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/StepTemplateConfig.java]: Documentation

File: build.gradle [hunk 3]
```diff
@@ -43,6 +43,7 @@ shadowJar {
         include(dependency('org.apache.httpcomponents:httpcore'))
         include(dependency('commons-logging:commons-logging'))
         include(dependency('com.google.guava:guava'))
+        include(dependency('org.jooq:joor-java-6'))
```
[the remark this answers is not in the document]
Comment (DzmitryHumianiuk) [reply build.gradle]: @pbortnik not sure we can use java8. we still have majority of users with java6 only.
@avarabyeu what do you think?

[the remark this answers is not in the document]
Comment (Yumfriez) [reply build.gradle]: Joor dependency removed. Java reflection used instead

File: src/main/java/com/epam/reportportal/annotations/Step.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/Step.java]: added

File: src/main/java/com/epam/reportportal/annotations/StepTemplateConfig.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/StepTemplateConfig.java]: added

File: src/main/java/com/epam/reportportal/annotations/Step.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/Step.java]: done

File: src/main/java/com/epam/reportportal/annotations/StepTemplateConfig.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/StepTemplateConfig.java]: added

**Pull request merged**